fundingRaised.py

```diff
@@ -1,134 +1,65 @@
-                    mapped['category'] = row[3]
-                    mapped['city'] = row[4]
-                    mapped['state'] = row[5]
-                    mapped['funded_date'] = row[6]
-                    mapped['raised_amount'] = row[7]
-                    mapped['raised_currency'] = row[8]
-                    mapped['round'] = row[9]
-                    return mapped
-
-        if 'city' in options:
-            for row in csv_data:
-                if row[4] == options['city']:
-                    mapped = {}
-                    mapped['permalink'] = row[0]
-                    mapped['company_name'] = row[1]
-                    mapped['number_employees'] = row[2]
-                    mapped['category'] = row[3]
-                    mapped['city'] = row[4]
-                    mapped['state'] = row[5]
-                    mapped['funded_date'] = row[6]
-                    mapped['raised_amount'] = row[7]
-                    mapped['raised_currency'] = row[8]
-                    mapped['round'] = row[9]
-                    return mapped
-
-        if 'state' in options:
-            for row in csv_data:
-                if row[5] == options['state']:
-                    mapped = {}
-                    mapped['permalink'] = row[0]
-                    mapped['company_name'] = row[1]
-                    mapped['number_employees'] = row[2]
-                    mapped['category'] = row[3]
-                    mapped['city'] = row[4]
-                    mapped['state'] = row[5]
-                    mapped['funded_date'] = row[6]
-                    mapped['raised_amount'] = row[7]
-                    mapped['raised_currency'] = row[8]
-                    mapped['round'] = row[9]
-                    return mapped
-
-        if 'round' in options:
-            for row in csv_data:
-                if row[9] == options['round']:
-                    mapped = {}
-                    mapped['permalink'] = row[0]
-                    mapped['company_name'] = row[1]
-                    mapped['number_employees'] = row[2]
-                    mapped['category'] = row[3]
-                    mapped['city'] = row[4]
-                    mapped['state'] = row[5]
-                    mapped['funded_date'] = row[6]
-                    mapped['raised_amount'] = row[7]
-                    mapped['raised_currency'] = row[8]
-                    mapped['round'] = row[9]
-                    return mapped
+    def find_by(options: dict) -> dict:
+        """
+        Returns first occurrence matching given options
+        """
+        columns, csv_data = load_data()
+
+        for row in csv_data:
+            options_match = []
+            for key, value in options.items():
+                if row[columns.index(key)] == value:
+                    options_match.append(True)
+                else:
+                    options_match.append(False)
+            if all(match for match in options_match):
```

Comment:
Duplicated code - the finding functionality is almost identical in both function. Could be somehow merged together in some utils